hazelcast/src/main/java/com/hazelcast/core/IMap.java

```diff
@@ -1329,4 +1330,28 @@ boolean tryLock(K key, long time, TimeUnit timeunit)
     <SuppliedValue, Result> Result aggregate(Supplier<K, V, SuppliedValue> supplier,
                                              Aggregation<K, SuppliedValue, Result> aggregation,
                                              JobTracker jobTracker);
+
+
+    /**
+     * Creates an always up to date snapshot of this {@code IMap}.
+     *
+     * @param name         the name of {@code QueryCache}
+     * @param predicate    the predicate for filtering entries.
+     * @param includeValue {@code true} if this {@code QueryCache} is allowed to cache values of entries, otherwise {@code false}
+     * @return the {@code QueryCache} instance with the supplied {@code name}
+     * @see QueryCache
+     */
+    QueryCache<K, V> getQueryCache(String name, Predicate<K, V> predicate, boolean includeValue);
```

Comment:
Please add documentation regarding null argument behaviour. The same goes for the method bellow.